File: ExpenseTracker.API/Models/Expense.cs
namespace ExpenseTracker.API.Models;

public class Expense
{
    public int Id { get; set; }
    public string Description { get; set; } = null!;
    public double Amount { get; set; }
    public DateTime Date { get; set; } = DateTime.Now;
    
    public string UserId { get; set; }
    public int CategoryId { get; set; }
    
    public User User { get; set; }
    public Category Category { get; set; }
}

File: expense-tracker-frontend/src/types/expense.ts
export type Expense = {
    id: number;
    categoryId: number;
    categoryName?: string;
    amount: number;
    description?: string | null;
    date: string;
};

export type CreateExpenseDto = {
    categoryId: number;
    amount: number;
    description?: string;
    date: string;
}

export type UpdateExpenseDto = CreateExpenseDto;

export type ExpenseChartPoint = {
    category: string;
    total: number;
}